server/handlerequest.go

```diff
@@ -25,37 +32,46 @@ func handleRequest(r *repo.Repo, handler Handler, jsonBytes []byte) (replyBytes
 		processingFunc()
 	}
 
+	// handle and process
 	switch handler {
 	case HandlerGetURIs:
 		getURIRequest := &requests.URIs{}
 		processIfJSONIsOk(json.Unmarshal(jsonBytes, &getURIRequest), func() {
 			reply = r.GetURIs(getURIRequest.Dimension, getURIRequest.IDs)
 		})
+		addMetrics(metrics, HandlerGetURIs, start, jsonErr, apiErr)
 	case HandlerGetContent:
 		contentRequest := &requests.Content{}
 		processIfJSONIsOk(json.Unmarshal(jsonBytes, &contentRequest), func() {
 			reply, apiErr = r.GetContent(contentRequest)
 		})
+		addMetrics(metrics, HandlerGetContent, start, jsonErr, apiErr)
 	case HandlerGetNodes:
 		nodesRequest := &requests.Nodes{}
 		processIfJSONIsOk(json.Unmarshal(jsonBytes, &nodesRequest), func() {
 			reply = r.GetNodes(nodesRequest)
 		})
+		addMetrics(metrics, HandlerGetNodes, start, jsonErr, apiErr)
 	case HandlerUpdate:
 		updateRequest := &requests.Update{}
 		processIfJSONIsOk(json.Unmarshal(jsonBytes, &updateRequest), func() {
 			reply = r.Update()
 		})
+		addMetrics(metrics, HandlerUpdate, start, jsonErr, apiErr)
 	case HandlerGetRepo:
 		repoRequest := &requests.Repo{}
 		processIfJSONIsOk(json.Unmarshal(jsonBytes, &repoRequest), func() {
 			reply = r.GetRepo()
 		})
+		addMetrics(metrics, HandlerGetRepo, start, jsonErr, apiErr)
 	default:
 		err = errors.New(log.Error("  can not handle this one " + handler))
 		errorResponse := responses.NewError(1, "unknown handler")
 		reply = errorResponse
+		addMetrics(metrics, "default", start, jsonErr, apiErr)
```

Comment:
Why don't you set metrics after the case? It's always the same including the "handler" from the switch